phys/module_bl_mynn.F

```diff
@@ -3634,11 +3677,18 @@ SUBROUTINE mynn_tendencies(kts,kte,      &
           ! Ice-friendly aerosols
           !=====================
           Dqnifa(k)=(qnifa2(k) - qnifa(k))/delt
+          IF (FLAG_QNBCA) THEN
+             !=====================
+             ! Black carbon aerosols
+             !=====================
+             Dqnbca(k)=(qnbca2(k) - qnbca(k))/delt
+          ENDIF
        ENDDO
     ELSE
        DO k=kts,kte
           Dqnwfa(k)=0.
           Dqnifa(k)=0.
+          Dqnbca(k)=0.
        ENDDO
```

Comment:
Do you want black carbon mixing to be done independently of the Thompson aerosols? Right now, the code looks like it can only produce a tendency for bca (lines 3680-3685) when you also produce tendencies for the aerosols.

Reply:
You're right-- BC will be done only when the other Thompson aerosols are computed. This is how we want it. The option will be either QNWFA+QNIFA only or QNWFA+QNIFA+QNBCA. Thanks Joe.